services/media/service/src/ingest/handlers/start-ingest-handler.ts

```diff
@@ -50,30 +71,92 @@ export class StartIngestHandler extends MediaGuardedTransactionalInboxMessageHan
   override async handleMessage(
     { payload, metadata }: TypedTransactionalMessage<StartIngestCommand>,
     ownerClient: ClientBase,
+    { subject }: GuardedContext,
   ): Promise<void> {
     // Sending only id in a scenario of detached services is an anti-pattern
     // Ideally the whole doc should have been sent and message should be self-contained, but because doc has a potential to be quite big - we save it to db and pass only it's id
-    const { document, id } = await selectExactlyOne('ingest_documents', {
-      id: payload.doc_id,
-    }).run(ownerClient);
-
-    const ingestItems: ingest_items.JSONSelectable[] = [];
-    for (const processor of this.entityProcessors) {
-      ingestItems.push(
-        ...(await this.initializeItems(
-          processor,
-          document.items,
-          id,
-          ownerClient,
-        )),
-      );
+    const { document, id, started_count } = await selectExactlyOne(
+      'ingest_documents',
+      { id: payload.doc_id },
+    ).run(ownerClient);
+
+    // Establish the order by which items of specific type would be processed,
+    // based on the order of entity processors.
+    const orderedTypes: Dict<number> = this.entityProcessors.reduce<
+      Dict<number>
+    >((result, value, index) => {
+      result[value.type] = index;
+      return result;
+    }, {});
+    const notStartedItems: IngestItem[] = document.items.sort(
+      (a, b) => orderedTypes[a.type] - orderedTypes[b.type],
```

Comment:
While my tests did produce consistent ordering results, It would be nice to have some explicit sorting guarantee. I think the second sort value should be `external_id`. Will try to implement this.